src/functional/functional.cpp

```diff
@@ -289,6 +289,38 @@ real2 SolutionIntegral<dim,nstate,real, MeshType>::evaluate_volume_integrand(
     return val;
 }
 
+template <int dim, int nstate, typename real, typename MeshType>
+OutletPressureIntegral<dim,nstate,real,MeshType>::OutletPressureIntegral(
+        std::shared_ptr<PHiLiP::DGBase<dim,real, MeshType>> dg_input,
+        const bool uses_solution_values,
+        const bool uses_solution_gradient)
+        : Functional<dim,nstate,real,MeshType>::Functional(dg_input,uses_solution_values,uses_solution_gradient)
+{}
+
+template <int dim, int nstate, typename real, typename MeshType>
+template <typename real2>
+real2 OutletPressureIntegral<dim,nstate,real,MeshType>::evaluate_boundary_integrand(
+        const PHiLiP::Physics::PhysicsBase<dim,nstate,real2> &physics,
+        const unsigned int                                    boundary_id,
+        const dealii::Point<dim,real2> &                      /*phys_coord*/,
+        const dealii::Tensor<1,dim,real2> &                    /*normal*/ ,
+        const std::array<real2,nstate> &                      soln_at_q,
+        const std::array<dealii::Tensor<1,dim,real2>,nstate> & /*soln_grad_at_q*/ ) const
+{
+    real2 pressure = 0;
+
+    if(boundary_id == 1002){
+        // casting it to a physics euler as it is needed for the pressure computation
+        const Physics::Euler<dim,nstate,real2>& euler_physics = dynamic_cast<const Physics::Euler<dim,nstate,real2>&>(physics);
+
+        // converting the state vector to the point pressure
+        pressure = euler_physics.compute_pressure(soln_at_q);
```

Comment:
Just a thought. Does `OutletPressureIntegral` work well for subsonic flows? As far as I understand, to get the goal oriented error in the functional, we 1. Solve the flow completely on coarse space and compute functional_coarse, 2. Solve the flow completely on fine space and compute functional_fine (this is approximated using taylor expansion) and 3. Take their difference. However, if we fix the back pressure as a BC (p_bc), then whether we compute the functional on fine or coarse space wouldn't matter and we would end up with the same p_bc at the outlet with no error. Not sure if the error indicator will work well in this case.

Reply:
Are you sure that the back pressure is fixed? When I tested it a few weeks ago, I think there was a small difference in the output functional when changing the Mach number, although it was small. In any case, I'm not actually using this for any test case so it's ok. The functional was originally implemented by Keigan in euler_gaussian_bump_adjoint.cpp (which no longer works), so maybe it can be useful for someone else in the future.